### PR TITLE
Implement welcome modal

### PR DESCRIPTION
**Pull request by gaoxk**
## Notion ticket link
https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=3714b8d75de041b683fa6ffc40bb4181
## Implementation description

- new modal

## Steps to test

1. login 
2. go to http://localhost:3000/#/?welcome=true
3. observe you can only close modal if you check box
4. go to http://localhost:3000/#/complete-profile
5. click submit button 
6. observe you are directed to go to http://localhost:3000/#/?welcome=true

## What should reviewers focus on?

- doth it work

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by Puepis — CHANGES_REQUESTED** [review]
everything works for me, the modal spacing looks a bit strange on my screen though e.g. the left side doesn't line up: 
image

can we get some designer feedback on the styling?

**Comment by gaoxk** [comment]
woops just realized i was missing the logo in the header too

File: frontend/src/components/homepage/WelcomeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    <Modal
+      isCentered
+      isOpen={isOpen}
+      closeOnOverlayClick={false}
+      onClose={onClose}
+      motionPreset="slideInBottom"
+      size="4xl"
+    >
+      <ModalOverlay />
+      <ModalContent marginTop="30vh">
```
Comment (Puepis) [inline frontend/src/components/homepage/WelcomeModal.tsx:43]: can we remove the `marginTop="30vh"` so that the modal is centered vertically?

**Review by Puepis — CHANGES_REQUESTED** [review]

**Review by Puepis — APPROVED** [review]
🚢 🚢 🚢

**Comment by gaoxk** [comment]
TFTR!

**Pull request merged**